beacon-chain/blockchain/service.go

```diff
@@ -75,17 +76,17 @@ func NewChainService(ctx context.Context, cfg *Config) (*ChainService, error) {
 
 // Start a blockchain service's main event loop.
 func (c *ChainService) Start() {
-	log.Info("Starting service")
-	var err error
-	c.genesisTime, err = c.beaconDB.GenesisTime()
-	if err != nil {
-		log.Fatalf("Unable to retrieve genesis time - blockchain service could not start: %v", err)
-		return
-	}
-	// TODO(#675): Initialize unfinalizedBlocks map from disk in case this
-	// is a beacon node restarting.
-	go c.updateHead(c.processedBlockChan)
-	go c.blockProcessing(c.processedBlockChan)
+	log.Info("Waiting for ChainStart log from the Validator Deposit Contract to start the beacon chain...")
+	subChainStart := c.web3Service.ChainStartFeed().Subscribe(c.genesisTimeChan)
```

Comment:
What if the chain has already started?

Reply:
+1 , add in a check for the chainservice